### DISCUSSION TITLE
St_AsGeoJSON wkid?

### DISCUSSION THREAD

**Discussion by DrewScatterday** [duckdb/duckdb-spatial#421] [Q&A] [answered]
Does the ST_AsGeoJSON function always output as wkid 4326?

**Comment by DrewScatterday** [comment]
and is there a way to make it return points back as a different wkid?

**Reply by DrewScatterday** [reply]
also does ST_GeomFromGeoJSON expect the geojson to be in 4326 wkid? or can i pass it wkid 3857 and then use a ST_Transform on it?

**Comment by Maxxen** [comment, marked as answer]
Hello! The `ST_AsGeoJSON` and `ST_GeomFromGeoJSON`, like most functions in DuckDB are agnostic the whatever SRID/WKID the input geometries are in, because DuckDB does not store nor track the SRID of individual geometries or columns. In this case you simply get out whatever you put in, the coordinate data itself just gets converted to/from string, but no other flipping/transformation takes place. If you want to transform the geometries into some other coordinate system you should use `ST_Transform` and provide both a source and target SRID.
